# # Definitions of tags to detect
#
# ## General remarks
#
# - All length in meters
# - Ellipsis (...) signifies that the previous element can be repeated multiple times.
#
# ## Standalone tag definitions
# ### Remarks
#
# - name is optional
#
# ### Syntax
#
# standalone_tags:
#   [
#     {id: ID, size: SIZE, name: NAME},
#     ...
#   ]
standalone_tags:
  [

    # Apriltags for 4 Way Intersction (1 in total)
    {id: 76, size: 0.065, name: TagA1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 74, size: 0.065, name: TagB1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 84, size: 0.065, name: TagC1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 85, size: 0.065, name: TagD1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 77, size: 0.065, name: TagE1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 41, size: 0.065, name: TagF1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 40, size: 0.065, name: TagG1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 75, size: 0.065, name: TagH1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    #Apriltags for 3 Way Intersections (4 in total)
    {id: 79, size: 0.065, name: TagP1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 79, size: 0.065, name: TagP2, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 79, size: 0.065, name: TagP3, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 79, size: 0.065, name: TagP4, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 83, size: 0.065, name: TagQ1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 83, size: 0.065, name: TagQ2, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 83, size: 0.065, name: TagQ3, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 83, size: 0.065, name: TagQ4, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 82, size: 0.065, name: TagR1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 82, size: 0.065, name: TagR2, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 82, size: 0.065, name: TagR3, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 82, size: 0.065, name: TagR4, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 80, size: 0.065, name: TagS1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 80, size: 0.065, name: TagS2, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 80, size: 0.065, name: TagS3, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 80, size: 0.065, name: TagS4, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 81, size: 0.065, name: TagT1, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 81, size: 0.065, name: TagT2, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 81, size: 0.065, name: TagT3, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 126, size: 0.065, name: TagT4, x: 0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

    {id: 78, size: 0.065, name: TagU1, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 78, size: 0.065, name: TagU2, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 86, size: 0.065, name: TagU3, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
    #{id: 78, size: 0.065, name: TagU4, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},

  ]

tag_bundles:
  [
  ]

  # # Definition of tag positions and orientations, defined as the world frame pose in the tag frame.
#standalone_tag_poses:
#  [
#
#    {id: 20, x: 0.3075, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5},
#    {id: 16, x: -0.3575, y: -0.0725, z: 0.585, qx: -0.5, qy: 0.5, qz: 0.5, qw: 0.5}

#  ]

intersection_tuples:
  [
    ['A','D'],['C','F'],['E','H'],['G','B'],['U','Q'],['P','S'],['R','T']
  ]
